### PR TITLE
[Doc] Add communication guide

### PR DESCRIPTION
**Pull request by joshuarrrr**
### Description

With info on slack, forum, and developer office hours.
Link from README, CONTRIBUTING, DEVELOPER_GUIDE

### Issues Resolved

N/A

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: COMMUNICATIONS.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# OpenSearch Dashboards Communication
+
+- [Overview](#overview)
+- [Slack](#slack)
+- [Forum](https://discuss.opendistrocommunity.dev/)
```
Comment (kavilla) [inline COMMUNICATIONS.md:5]: Nit: I think we have https://forum.opensearch.org/

**Review by kavilla — APPROVED** [review]
Awesome!

File: COMMUNICATIONS.md (hunk 1 above)
Comment (joshuarrrr) [reply COMMUNICATIONS.md:5]: nice catch - bad copy paste

Comment (joshuarrrr) [reply COMMUNICATIONS.md:5]: it's meant to link to the header section, anyway 😁

**Review by manasvinibs — APPROVED** [review]
Nice. LGTM

File: COMMUNICATIONS.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+Signing up isn't required to attend - all OpenSearch Dashboards contributors or interested developers are welcome as participants.
+
+Bring your ideas and projects early, while you still have time and flexibility to make significant changes.
+
+### When
+
+Every other Thursday, 10AM-11AM PT.
+
+### How to sign up
```
Comment (ashwin-pc) [inline COMMUNICATIONS.md:49]: Nit: A link to the forum here would be useful. This can just be a link to the Dashboard channel there for now.

**Review by ashwin-pc — APPROVED** [review]

**Review by manasvinibs — APPROVED** [review]

File: COMMUNICATIONS.md (hunk 2 above)
Comment (joshuarrrr) [reply COMMUNICATIONS.md:49]: Yeah, I'll add it once we have the forum posts. Some of this stuff is a little circular 😁

**Pull request merged**